client/trino-client/src/main/java/io/trino/client/OkHttpUtil.java

```diff
@@ -328,4 +331,13 @@ public static void setupKerberos(
         clientBuilder.addInterceptor(handler);
         clientBuilder.authenticator(handler);
     }
+
+    private static GSSContextProvider getDelegatedGSSContextProvider(Optional<GSSCredential> gssCredential)
+    {
+        if (gssCredential.isPresent()) {
```

Comment:
nit:  We can also use `map` and `orElse` to achieve the same

Reply:
Yep, but it requires cast:
`return gssCredential.map(gssCred -> (GSSContextProvider) new DelegatedConstrainedContextProvider(gssCred)).orElse(new DelegatedUnconstrainedContextProvider());`

If you think it's more readable, I can refactor

Reply:
refactored